File: neetcode.io/01-arrays/01-ContainsDuplicate/ts/solver.ts

// problem solver

function containsDuplicate(nums: number[]): boolean {
	function uniq(n: number[]): number[] {
		return Array.from(new Set(n))
	}
	const nums2 = uniq(nums);

	return nums.length != nums2.length;
};

export default {
	containsDuplicate
}

File: neetcode.io/01-arrays/01-ContainsDuplicate/go/solver.go
// Copyright (c) 2023 Furukawa, Atsushi <atsushifx@aglabo.com>
//
// This software is released under the MIT License.
// https://opensource.org/licenses/MIT

package main

// modules

// problem solver
func containsDuplicate(nums []int) bool {
	var nMap = map[int]int{}
	for i := 0; i < len(nums); i++ {
		nMap[nums[i]]++
		if nMap[nums[i]] >= 2 {
			return true
		}
	}
	return false
}
